src/sqltest/parts/connection/objectExplorerService.test.ts

```diff
@@ -753,4 +753,19 @@ suite('SQL Object Explorer Service tests', () => {
 			assert.equal(childNode.nodePath, objectExplorerExpandInfoRefresh.nodes[index].nodePath);
 		});
 	});
+
+	test('resolveTreeNodeChildren refreshes a node if it currently has an error', async () => {
+		await objectExplorerService.createNewSession('MSSQL', connection);
+		objectExplorerService.onSessionCreated(1, objectExplorerSession);
+
+		// If I call resolveTreeNodeChildren on a node with an error
+		let tablesNodePath = 'testServerName/tables';
+		let tablesNode = new TreeNode(NodeType.Folder, 'Tables', false, tablesNodePath, '', '', null, null, undefined, undefined);
+		tablesNode.connection = connection;
+		tablesNode.errorStateMessage = 'test error message';
+		await objectExplorerService.resolveTreeNodeChildren(objectExplorerSession, tablesNode);
```

Comment:
It might be a better test if resolveTreeNodeChildren has already been called on node previously and then error state is set on the node.